src/lib.rs

```diff
@@ -1918,6 +1932,40 @@ impl Connection {
         Ok(buf.len())
     }
 
+    /// Shuts down reading or writing from/to the specified stream.
+    ///
+    /// When the `direction` argument is set to [`Shutdown::Read`], outstanding
+    /// data in the stream's receive buffer is dropped, and no additional data
+    /// is added to it. Data received after calling this method is still
+    /// validated and ACKed but not stored, and [`stream_recv()`] will not
+    /// return it to the application.
+    ///
+    /// When the `direction` argument is set to [`Shutdown::Write`], outstanding
+    /// data in the stream's send buffer is dropped, and no additional data
+    /// is added to it. Data passed to [`stream_send()`] after calling this
+    /// method will be ignored.
```

Comment:
How does this work with recovery? Is unacked data still owned by the transport?

Reply:
Unacked data is part of the sent packets queue, so yes. When the packet gets acked we drop it along with the other frames in the packet, and if it doesn't get acked, when the packet is marked as lost we just drop the data anyway (i.e. we don't resend it).

Reply:
I think dropping send buffer data before getting ack is a little concerning unless we want to close the stream here. e.g. let's say we send a stream data, packet lost, we call shutdown(). Then client still try to send with lost packet in its ACK range. Since sender already dropped the stream data, so we don't retransmit as you say (do we send ack??), then client may keep requesting it until timed out.

Reply:
@junhochoi `shutdown()` should only be called if the application is not interested in the stream anymore. In the future we will need to add `STOP_SENDING` and `RESET_STREAM` so that this becomes more robust, but for now we will only use this when terminating requests (e.g. due to errors).